radis/tools/line_survey.py

```diff
@@ -263,24 +263,25 @@ def get_label_hitran(row, details):
 
         will be much faster!
         """
+        if "id" in attrs:
+            molecule = get_molecule(attrs["id"])
 
-        molecule = get_molecule(row.id)
+        else:
+            molecule = get_molecule(row.id)
 
         # Get global labels
         if molecule in HITRAN_CLASS1:
-            label = (
-                "{molec}[iso{iso:.0f}] [{branch}{jl:.0f}]({vl:.0f})->({vu:.0f})".format(
-                    **dict(
-                        [(k, row[k]) for k in ["vu", "vl", "jl", "iso"]]
-                        + [
-                            ("molec", molecule),
-                            ("branch", _fix_branch_format[row["branch"]]),
-                        ]
-                    )
+            label = "{molec} [{branch}{jl:.0f}]({vl:.0f})->({vu:.0f})".format(
```

Comment:
Keep this for later; but it's very important to keep the isotope in the Line-Survey. It's one of the things people want to know : which isotopes belong to this line ? 

Now, I know that the isotope may have been removed from the Line Database exported in the spectrum ; but in that case it should be in .conditions() 

We could do something like (pseudo-code below:): 
```python
try:
    iso = row["iso"] in row
except KeyError:
    if is_float(spectrum.conditions["isotope"]):  # make sure only one ? 
          iso  = ... 
    else:
          iso = '?'
dict_of_other_labels.update({"iso":iso}
```

Reply:
You can also use the attributes the Lines; see my comment above and suggestion to modifiy the Factory "get_lines" function.

Question : will the attributes be correctly serialized, i.e., preserved if you Store a spectrum on disk and reload ? 
```
s.store("my.spec")
s = radis.load_spec("my.spec")
s.lines.attrs["id"]      # does it work ?
```

Also, what if we merge two single-isotope spectra ? If lines are appended, an isotope column should be added. 
Keep this in a log note in your Issue ("things to check later"). How Line Survey and Lines are handled will change eventually : we'll store in the Spectrum a link to the database used rather than all of the lines. Less memory use!

Reply:
(for reference, link to our Slack discussion @gagan-aryan   https://radis-radiation.slack.com/archives/D01PB4L6GV6/p1624543791001300?thread_ts=1624483169.008800&cid=D01PB4L6GV6 )